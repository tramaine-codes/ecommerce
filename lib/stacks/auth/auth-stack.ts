import * as cdk from 'aws-cdk-lib';
import type * as cognito from 'aws-cdk-lib/aws-cognito';
import type { Construct } from 'constructs';
import type { Config } from '../../infrastructure/config/config.js';
import { ApiUserPool } from './pools/api-user-pool.js';

interface AuthStackProps extends cdk.StackProps {
  readonly config: Config;
}

export class AuthStack extends cdk.Stack {
  readonly apiUserPool: cognito.IUserPool;

  constructor(scope: Construct, id: string, props: AuthStackProps) {
    super(scope, id, props);

    this.apiUserPool = new ApiUserPool(this, props);
  }
}
