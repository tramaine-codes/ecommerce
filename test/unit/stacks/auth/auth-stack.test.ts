import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { expect, test } from 'vitest';
import { Config } from '../../../../lib/infrastructure/config/config.js';
import { AuthStack } from '../../../../lib/stacks/auth/auth-stack.js';

const app = new cdk.App();
const config = new Config();
const stack = new AuthStack(app, 'AuthStack', {
  config,
});

const template = Template.fromStack(stack);

test('matches the snapshot', () => {
  expect(template.toJSON()).toMatchSnapshot();
});
