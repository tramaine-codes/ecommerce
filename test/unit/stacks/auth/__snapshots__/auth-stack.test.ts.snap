// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "EcommerceApiPool917914B5": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AccountRecoverySetting": {
          "RecoveryMechanisms": [
            {
              "Name": "verified_phone_number",
              "Priority": 1,
            },
            {
              "Name": "verified_email",
              "Priority": 2,
            },
          ],
        },
        "AdminCreateUserConfig": {
          "AllowAdminCreateUserOnly": true,
        },
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "UserPoolName": "EcommerceApi",
        "VerificationMessageTemplate": {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Delete",
    },
    "EcommerceApiPoolEcommerceApiDomainE8D0F742": {
      "Properties": {
        "Domain": "ecommerce-api-tgillus",
        "UserPoolId": {
          "Ref": "EcommerceApiPool917914B5",
        },
      },
      "Type": "AWS::Cognito::UserPoolDomain",
    },
    "EcommerceApiPoolEcommerceApiResourceServer070841E7": {
      "Properties": {
        "Identifier": "EcommerceApiResourceService",
        "Name": "EcommerceApiResourceService",
        "Scopes": [
          {
            "ScopeDescription": "Ecommerce API read scope",
            "ScopeName": "ecommerceapi.read",
          },
          {
            "ScopeDescription": "Ecommerce API write scope",
            "ScopeName": "ecommerceapi.write",
          },
        ],
        "UserPoolId": {
          "Ref": "EcommerceApiPool917914B5",
        },
      },
      "Type": "AWS::Cognito::UserPoolResourceServer",
    },
    "EcommerceApiPoolTestPoolClientB1DAE862": {
      "Properties": {
        "AccessTokenValidity": 60,
        "AllowedOAuthFlows": [
          "client_credentials",
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": [
          {
            "Fn::Join": [
              "",
              [
                {
                  "Ref": "EcommerceApiPoolEcommerceApiResourceServer070841E7",
                },
                "/ecommerceapi.read",
              ],
            ],
          },
          {
            "Fn::Join": [
              "",
              [
                {
                  "Ref": "EcommerceApiPoolEcommerceApiResourceServer070841E7",
                },
                "/ecommerceapi.write",
              ],
            ],
          },
        ],
        "ClientName": "Test",
        "EnableTokenRevocation": true,
        "GenerateSecret": true,
        "RefreshTokenValidity": 1440,
        "SupportedIdentityProviders": [
          "COGNITO",
        ],
        "TokenValidityUnits": {
          "AccessToken": "minutes",
          "RefreshToken": "minutes",
        },
        "UserPoolId": {
          "Ref": "EcommerceApiPool917914B5",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
